src/config/session.c

```diff
@@ -118,14 +118,18 @@ read_environment_dir(const char *path_prefix)
 	DIR *envdir = opendir(path);
 
 	if (!envdir) {
-		if (errno != ENOENT) {
+		if (errno == ENOENT) {
+			wlr_log(WLR_INFO,
+				"no %s.d directory found",
+				path_prefix);
+		} else {
 			const char *err_msg = strerror(errno);
 			wlr_log(WLR_INFO,
 				"failed to read environment directory: %s",
 				err_msg ? err_msg : "reason unknown");
 		}
 
-		goto env_dir_cleanup;
+		goto no_dir_cleanup;
```

Comment:
Please keep the label the same; because we have more than one label in this file, I want to be as specific as possible about its purpose and think "env" is more helpful than "no" for this task.